### PR TITLE
add example to ExUnit.Assertions.assert_raise/2 doc

### PR DESCRIPTION
**Pull request by ctstacey**
adding an example of how to pass arguments to the function you want to assert raises an exception

### CODE REVIEW

File: lib/ex_unit/lib/ex_unit/assertions.ex [hunk 1]
```diff
@@ -755,6 +755,12 @@ defmodule ExUnit.Assertions do
         1 + "test"
       end
 
+  To pass arguments to `function`, wrap it:
```
Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/assertions.ex:758]: In this case, we are not really passing arguments to `function`. The function still receives 0 arguments. Rather, we are showing you can invoke any code you want inside `fn -> ... end`. Is this what you had in mind?

Comment (ctstacey) [reply lib/ex_unit/lib/ex_unit/assertions.ex:758]: Of course, thank-you! :)
Yes, that is what I had in mind.

File: lib/ex_unit/lib/ex_unit/assertions.ex [hunk 2]
```diff
@@ -755,6 +755,12 @@ defmodule ExUnit.Assertions do
         1 + "test"
       end
 
+  To test a function that requires arguments, wrap it:
+
+      assert_raise ArithmeticError, fn ->
+        my_add(1, "test")
+      end
+
```
Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/assertions.ex:763]: WDYT about

```suggestion
  Or if the anonymous function explicitly raises:

      assert_raise RuntimeError, fn ->
        raise "this assertion will pass due to this raise"
      end

```


?

Comment (ctstacey) [reply lib/ex_unit/lib/ex_unit/assertions.ex:763]: It's a good example of raising an exception, but doesn't answer the original question I had, _"How do I test my function that requires arguments?"_

Comment (josevalim) [reply lib/ex_unit/lib/ex_unit/assertions.ex:763]: Right. My concern is that there is nothing special to `assert_raise` as an answer to that question. You can invoke any code inside `fn -> ... end`, regardless if using `assert_raise` or not. So I was trying to find a way to add new information while answering your question, but I guess it was not enough. :)

Comment (ctstacey) [reply lib/ex_unit/lib/ex_unit/assertions.ex:763]: Yes, I see what you mean.
Let me think on it.

Comment (ctstacey) [reply lib/ex_unit/lib/ex_unit/assertions.ex:763]: Would this be too clunky?

```suggestion
  Calling the `raise/1` function:

      assert_raise RuntimeError, fn ->
        raise "assertion will pass due to this raise"
      end

```

File: lib/ex_unit/lib/ex_unit/assertions.ex (hunk 1 above)
Comment (ctstacey) [reply lib/ex_unit/lib/ex_unit/assertions.ex]: Thanks for your time and patience Jose, you're a hero!

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**